src/WikiService.Data/RubricRepository.cs

```diff
@@ -1,16 +1,91 @@
-﻿using LT.DigitalOffice.WikiService.Data.Interfaces;
+﻿using System.Linq;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+using LT.DigitalOffice.WikiService.Data.Interfaces;
 using LT.DigitalOffice.WikiService.Data.Provider;
+using LT.DigitalOffice.WikiService.Models.Db;
+using LT.DigitalOffice.WikiService.Models.Dto.Requests.Rubric.Filters;
 
 namespace LT.DigitalOffice.WikiService.Data
 {
   public class RubricRepository : IRubricRepository
   {
     private readonly IDataProvider _provider;
 
+    private IQueryable<DbRubric> CreateFindPredicates(
+      FindRubricFilter filter,
+      IQueryable<DbRubric> dbRubric)
+    {     
+      if (!string.IsNullOrEmpty(filter.NameIncludeSubstring))
+      {
+        dbRubric = dbRubric.Where(
+          rubric =>
+            rubric.Name.Contains(filter.NameIncludeSubstring.ToLower()));
+      }
+
+      if (filter.IsAscendingSort.HasValue)
+      {        
+        dbRubric = filter.IsAscendingSort.Value
+          ? dbRubric.OrderBy(rubric => rubric.Name)
+          : dbRubric.OrderByDescending(rubric => rubric.Name);
+      }
+
+      if (filter.IsActive.HasValue)
+      {
+        dbRubric = dbRubric.Where(rubric => rubric.IsActive == filter.IsActive.Value);
+      }
+      else
+      {
+        dbRubric = dbRubric.OrderByDescending(rubric => rubric.CreatedAtUtc);
+      }
+
+      foreach(DbRubric topRubric in dbRubric.ToList())
```

Comment:
May be replace it to another method? Because break exits only once from the inner foreach. And because of the outer foreach, the process will go further.